### DISCUSSION TITLE
Http::fake does not allow wildcards as parameters

### DISCUSSION THREAD

**Discussion by stephan-v** [laravel/framework#46572] [Ideas]
It seems that `Http::fake` does not allow a wildcard as "route parameter". A wildcard only works at the end of the string.

An example:

```
Http::fake([
    '/accounts/*' => Http::response(), // This works fine.
    '/users/*/settings' => Http::response(), // This doesn't work.
])->baseUrl($url);
```

The route I want to mock in this example would be something like `/users/8665/settings`.

In this case where you would also want to mock a route like `/users/8665/likes` you can't just use `/users/*` to fix the example because this would then be applicable to both, which is unwanted. With nested routes it is pretty easy to run into this scenario. 

Would there be a good reason not to allow this?

**Comment by shaedrich** [comment]
It would be cool, if this could make use of [URI templates](https://www.rfc-editor.org/rfc/rfc6570) as introduced into Laravel 9 via #45982
